src/shared/bus-polkit.c

```diff
@@ -299,6 +366,29 @@ static int process_polkit_response(
         return -EACCES;
 }
 
+static AsyncPolkitQuery *find_query(Hashmap *registry, sd_bus_message *call, const char *action, AsyncPolkitQueries **ret_queries) {
+        AsyncPolkitQueries *qs;
+        AsyncPolkitQuery *q = NULL;
+
+        assert(call);
+        assert(action);
+        assert(ret_queries);
+
+        qs = hashmap_get(registry, call);
+        if (!qs)
+                return NULL;
+
+        LIST_FOREACH(item, i, qs->items)
```

Comment:
How about to store query into a hashmap by the action?

Reply:
It's not worth it. In most cases there'll be only one item in the list.